File: app/src/main/java/com/ht117/movierank/favorites/FavoritesInteractor.java
package com.ht117.movierank.favorites;


import com.ht117.movierank.Movie;

import java.util.List;

/**
 * @author arun
 */
public interface FavoritesInteractor
{
    void setFavorite(Movie movie);
    boolean isFavorite(String id);
    List<Movie> getFavorites();
    void unFavorite(String id);
}

File: app/src/main/java/com/ht117/movierank/favorites/FavoritesInteractor.kt
package com.ht117.movierank.favorites


import com.ht117.movierank.model.Movie

/**
 * @author Quang Pham
 */
interface FavoritesInteractor {
    val favorites: List<Movie>
    fun setFavorite(movie: Movie)
    fun isFavorite(id: String): Boolean
    fun unFavorite(id: String)
}
